elyra/pipeline/validation.py

```diff
@@ -165,17 +165,32 @@ def _validate_pipeline_structure(self, pipeline: dict, response: ValidationRespo
         elif 'version' not in primary_pipeline['app_data']:
             response.add_message(severity=ValidationSeverity.Error, message_type="invalidPipeline",
                                  message="Pipeline app_data 'version' is missing from primary pipeline")
-        elif not int(primary_pipeline['app_data']['version']) > 0:
+
+        # validate pipeline version compatibility
+        pipeline_version = int(primary_pipeline['app_data']['version'])
+        if pipeline_version not in range(PIPELINE_CURRENT_VERSION + 1):
```

Comment:
`range(PIPELINE_CURRENT_VERSION + 1)` -> 0 ... 5. Was there ever a version 0?

Reply:
The first pipelines will have no version field, which will be 0 with the updated code above.

Reply:
hmmm, if there is no `version` property defined, `primary_pipeline['app_data']['version']` would fail, no?